pybaseball/split_stats.py

```diff
@@ -0,0 +1,125 @@
+import requests
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+import bs4 as bs
+import pandas as pd
+import re
+
+
+def download_url(url):
+    """
+    Gets the content from the url specified
+    """
+    session = requests.Session()
+    retry = Retry(connect=3, backoff_factor=0.5)
+    adapter = HTTPAdapter(max_retries=retry)
+    session.mount('http://', adapter)
+    session.mount('https://', adapter)
+    resp = session.get(url)
+    return resp.content
+
+
+def get_split_soup(playerid, year=None, pitching_splits=False):
+    """
+    gets soup for the player splits. 
+    """
+    if pitching_splits is True:
+        pitch_or_bat = 'p'
+    else:
+        pitch_or_bat = 'b'
+    if year is None:  # provides scores from yesterday if date is not provided
+        url = f"https://www.baseball-reference.com/players/split.fcgi?id={playerid}&year=Career&t={pitch_or_bat}"
+    else:
+        year = str(year)
+        url = f"https://www.baseball-reference.com/players/split.fcgi?id={playerid}&year={year}&t={pitch_or_bat}"
+    html = download_url(url)
+    soup = bs.BeautifulSoup(html, 'lxml')
+    return soup
+
+
+def get_player_info(playerid=None, soup=None):
+    '''
+    Returns a dictionary with player position, batting and throwing handedness, player height in inches, player weight, and current team from Baseball Reference. 
+    '''
+
+    if not soup:
+        soup = get_split_soup(playerid)
+    about_info = soup.find_all(
+        "div", {"itemtype": "https://schema.org/Person"})
+    info = [ele for ele in about_info]
+    for i in range(len(info)):
+        ptags = info[i].find_all('p')
+        fv = []
+        for j in range(len(ptags)):
+            InfoRegex = re.compile(r'>(.*?)<', re.DOTALL)
+            r = InfoRegex.findall(str(ptags[j]))
+            for k in range(len(r)):
+                pattern = re.compile('[\W_]+')
+                strings = pattern.sub(' ', r[k])
+                if strings and strings != ' ':
+                    fv.append(strings)
```

Comment:
The last thing to cover before this is ready, I think:
Can you provide a comment for this one block of code? Everything else is fairly self explanatory, this is the one thing I don't really understand here.

Reply:
done